### PR TITLE
util/deephash: add packet filter benchmark

### PR DESCRIPTION
**Pull request by bradfitz**
(breaking up parts of another change)

This adds a PacketFilter hashing benchmark with an input that both
contains every possible field, but also is somewhat representative in
the shape of what real packet filters contain.

### CODE REVIEW

File: util/deephash/deephash_test.go [hunk 1]
```diff
@@ -324,6 +324,55 @@ func BenchmarkHash(b *testing.B) {
 	}
 }
 
+func ptrTo[T any](v T) *T { return &v }
+
+// filterRules is a 1-element FilterRule with everything populated.
```
Comment (maisem) [inline util/deephash/deephash_test.go:329]: 1-element?

**Review by maisem — APPROVED** [review]
Results?

File: util/deephash/deephash_test.go (hunk 1 above)
Comment (bradfitz) [reply util/deephash/deephash_test.go]: Updated comment to more match the commit message.

**Comment by bradfitz** [comment]
> Results?

Not interesting at this point. I just want this in for the upcoming optimization to compare against.

**Pull request merged**